EIPS/eip-5XXX.md

```diff
@@ -0,0 +1,341 @@
+    - name: amount
+      indexed: false
+      type: uint256
+```
+
+#### Mature
+_Alberto: Maybe we need this event, even if we don't have a `mature` function._
+
+The Principal Token has hit maturity, and a user has called `redeem` or `mature` on the Principal Token contract for the first time post-maturity.
+
+OPTIONAL - This event can be used to improve usability and clarity for index based systems, but interfaces and other contracts MUST NOT expect this event to be present.
+
+MUST be emitted when the `mature` method is called and the `maturityRate` is stored.
+
+```yaml
+- name: Mature
+  type: event
+
+  inputs:
+    - name: timestamp
+      indexed: true
+      type: uint256
+    - name: exchangeRate
+      indexed: true
+      type: uint256
+```
+
+## Rationale
+
+The Principal Token interface is designed to be optimized for integrators with a core minimal interface alongside optional interfaces to enable backwards compatability. Details such as accounting and allocation of deposited tokens are intentionally not specified, as Principal Tokens are expected to be treated as black boxes on-chain and inspected off-chain before use.
+
+ERC-20 is enforced as implementation details such as token approval and balance calculation directly carry over. This standardization makes Principal Tokens immediately compatible with all ERC-20 use cases in addition to ERC-5070.
+
+The optional `redeem` method is included given an ideal Principal Token implementation includes all necessary methods for primary use-cases. All principal tokens are redeemable upon maturity, with the only variance being whether further yield is generated post-maturity. Given the ubiquity of redemption, the presence of `redeem` allows integrators to purchase Principal Tokens on an open market, and them later redeem them for a fixed-yield solely knowing the address of the Principal Token itself.
+
+The optional `mint` method was also included to similarly cover the creation of Principal Tokens. Though a ubiquitous pattern for minting is less present in current implementations, and is less necessary than `redemption` for common use cases, the ability to directly mint Principal Tokens reduces the complexity of integrations.
+
+The optional `maturityRate` and `exchangeRate` methods are included to allow for the accommodation of additional yield generation post-maturity. In implementations that include these optional methods, the amount of underlying returned by the `redeem` method is equivalent to ```amount + (amount * (exchangeRate/maturityRate))```.
+
+## Backwards Compatibility
+
+ERC-5070 is fully backward compatible with the ERC-20 standard and has no known compatibility issues with other standards.
+For production implementations of Principal Tokens which do not use ERC-5070, wrapper adapters can be developed and used, or wrapped tokens can be implemented.
+
+## Reference Implementations
+
+Official Example Implementation: https://github.com/JTraversa/Principal-Token-Standard
```

Comment:
No external links.  Either inline a reference implementation, or add one to `../assets/eip-5089/*`.  This section is optional if you can't craft a small reference implementation inline or you want to wait to add it until later.

Reply:
Same comment for other links.

Reply:
Noted, thank you.

Reply:
Fixed on [PR5095](https://github.com/ethereum/EIPs/pull/5095)